src/parser/test/test.ml

```diff
@@ -0,0 +1,9262 @@
+        List
+          ([ Atom "@param"; Atom s ] @ List.map (at nestable_block_element) es)
+    | `Raise (s, es) ->
+        List
+          ([ Atom "@raise"; Atom s ] @ List.map (at nestable_block_element) es)
+    | `Return es ->
+        List (Atom "@return" :: List.map (at nestable_block_element) es)
+    | `See (kind, s, es) ->
+        let kind =
+          match kind with
+          | `Url -> "url"
+          | `File -> "file"
+          | `Document -> "document"
+        in
+        List
+          ([ Atom "@see"; Atom kind; Atom s ]
+          @ List.map (at nestable_block_element) es)
+    | `Since s -> List [ Atom "@since"; Atom s ]
+    | `Before (s, es) ->
+        List
+          ([ Atom "@before"; Atom s ] @ List.map (at nestable_block_element) es)
+    | `Version s -> List [ Atom "@version"; Atom s ]
+    | `Canonical p -> List [ Atom "@canonical"; at str p ]
+    | `Inline -> Atom "@inline"
+    | `Open -> Atom "@open"
+    | `Closed -> Atom "@closed"
+
+  let block_element : Ast.block_element -> sexp = function
+    | #Ast.nestable_block_element as e -> nestable_block_element e
+    | `Heading (level, label, es) ->
+        let label = List [ Atom "label"; opt str label ] in
+        let level = string_of_int level in
+        List [ Atom level; label; List (List.map (at inline_element) es) ]
+    | `Tag t -> tag t
+
+  let docs : Ast.docs -> sexp = fun f -> List (List.map (at block_element) f)
+end
+
+let error err = Atom (Odoc_parser.Error.to_string err)
+
+let parser_output formatter { Odoc_parser.Error.value; warnings } =
+  let value = Ast_to_sexp.docs value in
+  let warnings = List (List.map error warnings) in
+  let output =
+    List [ List [ Atom "output"; value ]; List [ Atom "warnings"; warnings ] ]
+  in
+  Sexplib0.Sexp.pp_hum formatter output;
+  Format.pp_print_flush formatter ()
+
+let test ?(location = { Location.line = 1; column = 0 }) str =
+  let dummy_filename = "f.ml" in
+  let location =
+    {
+      Lexing.pos_fname = dummy_filename;
+      pos_lnum = location.line;
+      pos_bol = 0;
+      pos_cnum = location.column;
+    }
+  in
+  let ast = Odoc_parser.parse_comment ~location ~text:str in
+  Format.printf "%a" parser_output ast
+
+let%expect_test _ =
+  let module Trivial = struct
+    let empty =
+      test "";
+      [%expect "((output ()) (warnings ()))"]
+
+    let space =
+      test " ";
+      [%expect "((output ()) (warnings ()))"]
+
+    let two_spaces =
+      test "  ";
+      [%expect "((output ()) (warnings ()))"]
+
+    let tab =
+      test "\t";
+      [%expect "((output ()) (warnings ()))"]
+
+    let mixed_space =
+      test " \t \t";
+      [%expect "((output ()) (warnings ()))"]
+
+    let newline =
+      test "\n";
+      [%expect "((output ()) (warnings ()))"]
+
+    let blank_line =
+      test "\n\n";
+      [%expect "((output ()) (warnings ()))"]
+
+    let cf_lf =
+      test "\r\n";
+      [%expect "((output ()) (warnings ()))"]
+  end
+  [@ocaml.warning "-32"] in
```

Comment:
Apparently ocaml `<= 4.05` doesn't handle these attributes. I suggest removing that warning at the toplevel, just before the first test suite:
```ocaml
[@@@ocaml.warning "-32"]
```